packages/zwave-js/src/lib/driver/Driver.ts

```diff
@@ -2412,6 +2414,56 @@ ${handlers.length} left`,
 				if (!msg.command.moreUpdatesFollow) {
 					nodeSessions.supervision.delete(msg.command.sessionId);
 				}
+			} else if (
```

Comment:
I think handling Supervision Get should be part of the `else` branch because the `awaitedCommands` logic should be applied too.

And we should probably return a status of `Fail` when there's an error during `handleCommand`

Reply:
Does the `awaitedCommands` logic also eventually call back to `handleRequest`? If the return on line 2515 is called could we leave the method before sending back a report? Or should we send back a success report after resolving the promise and before returning?

How would we know if there was an error during `handleCommand`? It returns void and I can't see anything obvious that throws.

Reply:
> Does the awaitedCommands logic also eventually call back to handleRequest? 

No, that is for out-of-flow logic like bootstrapping processes to be able to wait for specific commands. The request is handled by the waiting code. In that case, just send success back.

> How would we know if there was an error during handleCommand?

There shouldn't be anything obvious, but there could be unexpected errors. Just `try/catch` it, send Fail back and rethrow the error when you hit the catch branch. I don't expect this to be a frequent thing, but telling the device we couldn't handle the command seems like the logical thing to do.

Reply:
Makes sense! I've made those changes now - look okay?

Reply:
It's a bit inelegant, but I'll take it from here. Thanks so far!